libraries/Microsoft.Bot.Builder.LanguageGeneration/Template.cs

```diff
@@ -20,34 +17,59 @@ public class Template
         /// <summary>
         /// Initializes a new instance of the <see cref="Template"/> class.
         /// </summary>
-        /// <param name="parseTree">The parse tree of this template.</param>
-        /// <param name="lgfileContent">lg file content.</param>
+        /// <param name="templateName">Template name without parameters.</param>
+        /// <param name="parameters">Parameter list.</param>
+        /// <param name="templateBody">Template content.</param>
+        /// <param name="startLine">StartLine of template (zero based).</param>
+        /// <param name="stopLine">Stop line of template (zero based).</param>
         /// <param name="source">Source of this template.</param>
-        internal Template(LGFileParser.TemplateDefinitionContext parseTree, string lgfileContent, string source = "")
+        internal Template(
+            string templateName,
+            List<string> parameters,
+            string templateBody,
+            int startLine,
+            int stopLine,
+            string source = "")
         {
-            ParseTree = parseTree;
-            Source = source;
-
-            Name = ExtractName();
-            Parameters = ExtractParameters();
-            Body = ExtractBody(lgfileContent);
+            this.Name = templateName ?? string.Empty;
+            this.Parameters = parameters ?? new List<string>();
+            this.Body = templateBody ?? string.Empty;
+            this.StartLine = startLine;
```

Comment:
Probably LineStart, LineEnd?

And can we merge source and line start,end to one single structure?